apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/IgnoredTracerContext.java

```diff
@@ -34,14 +34,21 @@ public class IgnoredTracerContext implements AbstractTracerContext {
     private static final NoopSpan NOOP_SPAN = new NoopSpan();
     private static final String IGNORE_TRACE = "Ignored_Trace";
 
+    private LinkedList<AbstractSpan> activeSpanStack;
```

Comment:
Why need this?

Reply:
Because the old span is still in use, if it's discarded, [span#prepareForAsync](https://github.com/apache/skywalking-java/blob/1a010470076ba7f206f4fd6559d04136bea73417/apm-sniffer/optional-plugins/optional-spring-plugins/optional-spring-cloud/gateway-2.0.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/spring/cloud/gateway/v20x/NettyRoutingFilterInterceptor.java#L45) will throw an error.

Reply:
![image](https://github.com/apache/skywalking-java/assets/16240434/e0cc5df8-a63c-4504-a1ea-15d54b126fcd)
![image](https://github.com/apache/skywalking-java/assets/16240434/fcc0ecbb-31da-4799-8934-9b50aa4370e8)
![image](https://github.com/apache/skywalking-java/assets/16240434/7e66c4c4-0882-4485-b4b1-1b40bb6abaf3)

Reply:
This is not friendly for GC. We could add `ingored` flag to the span and skip this check.

Reply:
Another option maybe to create a ForceIgnoreTracerContext to deal with created span cases. 
And this only happens in very rare case, so we should process that only when we checked all active spans, which have one running in async mode.

Reply:
Do you prefer replacing the span's context with IgnoredTracerContext for old spans with old [context references](https://github.com/apache/skywalking-java/blob/4dfc1e85b56e75a1bf2bae11d34d8c0fe234bed0/apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/trace/AbstractTracingSpan.java#L68), or would you rather ignore any operations involving context within the span?

Reply:
I didn't dig in so deeply, as I am feeling this change is not easy.

I just provided two possibilities, no preference. One key part is, what you are adding is not happening as always, you should not make the agent kernel costs too much.

I don't whether this is possible.

Reply:
That was why I said, this could be done easily if this only happens in your private fork. But in general, I don't have a good idea could be done easily. 

But this has to be very good, as your changes are going to affect every user for sure.